### PR TITLE
Marchuk

### PR DESCRIPTION
**Pull request by EvgenyMarchuk**
Added test test method StatusCodeTest.

### CODE REVIEW

File: src/main/java/core/BmpTestBase.java [hunk 1]
```diff
@@ -41,4 +42,8 @@ public void cleanup() {
         saveEnvironment();
     }
 
+
+    public ProxyServer getServer() {
```
Comment (Kreisfahrer) [inline src/main/java/core/BmpTestBase.java]: Just make server field protected - when there is an inheritance it is more logical

File: src/test/java/StatusCodesTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    @BeforeMethod
+    public void setup(){
+        bmp = getServer();
+        open("status_codes");
+
+    }
+
+    @Test
+    public void responseCodeTest(){
+        List<String> links = getUrls($$(".example>ul>li>a"), "href");
```
Comment (Kreisfahrer) [inline src/test/java/StatusCodesTest.java]: you should use one of the page object variants

File: src/main/java/helpers/Helpers.java [hunk 3]
```diff
@@ -1,14 +1,26 @@
...
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class Helpers {
     public static List<String> readAllLines(Path resourcePath) throws IOException {
         return Files.readAllLines(resourcePath, Charset.defaultCharset());
     }
+
+    public static List<String> getUrls(ElementsCollection elements, String attribute) {
```
Comment (Kreisfahrer) [inline src/main/java/helpers/Helpers.java:19]: You won't need this method if you just click the link

File: src/test/java/StatusCodesTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        Map<String, String> responseCode = new HashMap<>();
+        for (String link : links) {
+            bmp.newHar("the-internet");
+            open(link);
+            Har har = bmp.getHar();
+            for (HarEntry entry : har.getLog().getEntries()) {
```
Comment (Kreisfahrer) [inline src/test/java/StatusCodesTest.java:32]: You should get hard once in test (use click on every link then get hard)

File: src/test/java/StatusCodesTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        bmp = getServer();
+        open("status_codes");
+
+    }
+
+    @Test
+    public void responseCodeTest(){
+        List<String> links = getUrls($$(".example>ul>li>a"), "href");
+        Map<String, String> responseCode = new HashMap<>();
+        for (String link : links) {
```
Comment (Kreisfahrer) [inline src/test/java/StatusCodesTest.java]: you need to find a way to visit every link in cycle (with click() and back())

File: src/main/java/core/BmpTestBase.java (hunk 1 above)
Comment (EvgenyMarchuk) [inline src/main/java/core/BmpTestBase.java]: Corrected

**Comment by EvgenyMarchuk** [comment]
Closed pull

**Pull request closed without merging**